apps/dashboard/app/helpers/batch_connect/session_contexts_helper.rb

```diff
@@ -1,7 +1,7 @@
 # Helper for creating new batch connect sessions.
 module BatchConnect::SessionContextsHelper
-  def create_widget(form, attrib, format: nil, hide_excludable: true)
-    return '' if attrib.fixed?
+  def create_widget(form, attrib, format: nil, hide_excludable: true, hide_fixed: true)
+    return '' if hide_fixed && attrib.fixed?
```

Comment:
I think instead of a decorator we should just provide a partial here for fixed attributes.

Reply:
> As far as I know, the problem is that Script fixed attributes need to be rendered differently from BatchConnect attributes.

I'm not sure I follow you on the issue you're trying to articulate maybe I need to look more closely at what the decorator is being used for.  But I guess I was suggesting a new partial for just show (as the create_widget api is used for show).

Reply:
If this what you had in mind?
```ruby
if attrib.fixed?
      render :partial => attrib.fixed_partial, :locals => { form: form, attrib: attrib, field_options: field_options }
end
```

Reply:
sorta yea - like `path_selector` or `file_attachments`.

Only `hide_fixed` should default to false (to preserve batch connect behavior) 

```ruby
if attrib.fixed?
  hide_fixed ? render(partial: 'some_new_partial, ...) : ''
end
```

Reply:
> Only hide_fixed should default to false (to preserve batch connect behavior)

Sorry - you've got the correct default - I get boolean expressions mixed up some times.

Reply:
I committed  the changes, it will make it easier to visualize the issue

Reply:
> As the edit view is using as well create_widget, it now renders the readonly view of the element, the same as in show.

edit views should be using `create_editable_widget` which I see you have updates for. I'll have to pull it down and hack around to get a sense of the problem space you're describing.

Reply:
> Should I continue in this direction?

I think this is the right direction yes. I think the only issue I have now is one of maintenance. How to achieve this in a way that will continue to make sense in 3 months. 

Specifically the updates in `create_editable_widget`. Maybe just a comment will suffice? I think what you have is basically as simple as it can be (simplest being the best), so a a short comment on _why_ may just help us remember why we're modifying the attribute in a view.